utils/check_repo.py

```diff
@@ -646,6 +647,31 @@ def check_all_auto_object_names_being_defined():
         raise Exception(f"There were {len(failures)} failures:\n" + "\n".join(failures))
 
 
+def check_all_auto_mapping_names_in_config_mapping_names():
+    """Check all keys defined in auto mappings (mappings of names) appear in `CONFIG_MAPPING_NAMES`."""
+    failures = []
+
+    # `TOKENIZER_PROCESSOR_MAPPING_NAMES` and `AutoTokenizer` is special, and don't need to follow the rule.
+    mapping_to_check = {
```

Comment:
`AutoTokenizer` can check `tokenizer_type` which makes it special
```python
        tokenizer_type = kwargs.pop("tokenizer_type", None)
        # First, let's see whether the tokenizer_type is passed so that we can leverage it
        if tokenizer_type is not None:
            tokenizer_class = None
            tokenizer_class_tuple = TOKENIZER_MAPPING_NAMES.get(tokenizer_type, None)
```